packages/lab/src/overlay/Overlay.css

```diff
@@ -1,93 +1,47 @@
+  font-size: var(--saltOverlay-fontSize, var(--salt-text-fontSize));
+  font-weight: var(--saltOverlay-fontWeight, var(--salt-text-fontWeight));
+  line-height: var(--saltOverlay-lineHeight, var(--salt-text-lineHeight));
 
-.saltOverlay-content {
-  display: flex;
-  padding: calc(var(--overlay-spacing) * 2);
-  font-size: var(--overlay-fontSize);
+  border-color: var(--overlay-borderColor);
+  border-style: var(--saltOverlay-borderStyle, var(--salt-container-borderStyle));
+  border-width: var(--saltOverlay-borderWidth, var(--salt-size-border));
 
-  border: 1px solid var(--overlay-borderColor);
-  box-shadow: var(--salt-overlayable-shadow-popout);
-  line-height: var(--salt-text-lineHeight);
   background: var(--overlay-background);
-  color: var(--overlay-text-color);
-  position: relative;
-  opacity: 1;
-  overflow: visible;
-  z-index: var(--overlay-zindex);
-}
-
-.saltOverlay-arrow,
-.saltOverlay-arrow::after {
-  border: 8px solid transparent;
-  position: absolute;
-}
-
-.saltOverlay-arrow::after {
-  content: "";
-}
-
-.saltOverlay[data-placement^="top"] .saltOverlay-arrow {
-  border-top-color: var(--overlay-borderColor);
-  bottom: -17px;
-}
-
-.saltOverlay[data-placement^="bottom"] .saltOverlay-arrow {
-  border-bottom-color: var(--overlay-borderColor);
-  top: -17px;
-}
-
-.saltOverlay[data-placement^="left"] .saltOverlay-arrow {
-  border-left-color: var(--overlay-borderColor);
-  right: -17px;
-}
-
-.saltOverlay[data-placement^="right"] .saltOverlay-arrow {
-  border-right-color: var(--overlay-borderColor);
-  left: -17px;
-}
+  box-shadow: var(--saltOverlay-boxShadow, var(--salt-overlayable-shadow-popout));
 
-.saltOverlay[data-placement^="top"] .saltOverlay-arrow::after {
-  border-top-color: var(--overlay-background);
-  bottom: -7px;
-  left: calc(100% - 8px);
+  color: var(--saltOverlay-text-color, var(--salt-text-primary-foreground));
+  z-index: var(--saltOverlay-zIndex, var(--salt-zIndex-flyover));
+  width: max-content;
 }
 
-.saltOverlay[data-placement^="bottom"] .saltOverlay-arrow::after {
-  border-bottom-color: var(--overlay-background);
-  top: -7px;
-  left: calc(100% - 8px);
+.saltOverlayPanelBase-container {
+  animation: fade-in var(--salt-duration-perceptible) ease-in-out;
+  position: relative;
 }
 
-.saltOverlay[data-placement^="left"] .saltOverlay-arrow::after {
-  border-left-color: var(--overlay-background);
-  right: -7px;
-  top: calc(100% - 8px);
+@keyframes fade-in {
+  0% {
+    opacity: 0;
+  }
+  100% {
+    opacity: 1;
+  }
 }
 
-.saltOverlay[data-placement^="right"] .saltOverlay-arrow::after {
-  border-right-color: var(--overlay-background);
-  left: -7px;
-  top: calc(100% - 8px);
+.saltOverlayPanelBase-content {
+  overflow: auto;
+  padding: var(--saltOverlay-content-padding, var(--salt-spacing-100) 0 var(--salt-spacing-100) var(--salt-spacing-100));
 }
 
-/* Close Button */
-.saltOverlay-close {
-  position: absolute;
-  right: 0;
+.saltButton-secondary.saltOverlayPanelBase-closeButton {
```

Comment:
minor: is the increased specificity needed ? Can ".saltButton.secondary" be removed ?

Reply:
we need it to ensure button position remains sticky when content is longer than overlay panel